.github/actions/bouf/action.yaml

```diff
@@ -0,0 +1,122 @@
+      shell: pwsh
+      run: |
+        # Extract artifact in build dir and remove zip
+        cd build
+        Expand-Archive -Path *.zip -DestinationPath .
+        del *.zip
+
+    - name: Setup bouf
+      shell: pwsh
+      env:
+        BOUF_VERSION: 'v0.6.0'
+        BOUF_HASH: '06861ae6d714b754602000bd51607b5b72e655b978b7e7497b0eb970b1613dd3'
+        BOUF_NSIS_HASH: 'bb0561a76cecaba62785642dd71052a9fe53458e45df108ca08867ee333b80bd'
+      run: |
+        # Download bouf release
+        mkdir bouf && cd bouf
+        
+        curl -o bouf-windows.zip -L https://github.com/obsproject/bouf/releases/download/$Env:BOUF_VERSION/bouf-windows-$Env:BOUF_VERSION.zip
+        if ((Get-FileHash bouf-windows.zip -Algorithm SHA256).Hash -ne "$Env:BOUF_HASH") {
+          throw "bouf hash does not match."
+        }
+        Expand-Archive -Path bouf-windows.zip -DestinationPath bin
+        
+        # Download release's NSIS data (script + assets)
+        curl -o bouf-nsis.zip -L https://github.com/obsproject/bouf/releases/download/$Env:BOUF_VERSION/bouf-nsis-$Env:BOUF_VERSION.zip
+        if ((Get-FileHash bouf-nsis.zip -Algorithm SHA256).Hash -ne "$Env:BOUF_NSIS_HASH") {
+          throw "NSIS package hash does not match."
+        }
+        Expand-Archive -Path bouf-nsis.zip -DestinationPath nsis
+
+    - name: Download Google CNG Provider
+      shell: pwsh
+      env:
+        CNG_TAG: 'cng-v0.9'
+        GH_TOKEN: ${{ github.token }}
+      run: |
+        # Download Google CNG provider release from github
+        mkdir gcng && cd gcng
+        gh release download $env:CNG_TAG -R "GoogleCloudPlatform/kms-integrations" -p "*amd64.zip"
+        Expand-Archive -Path *.zip -DestinationPath tmp
+        Get-ChildItem tmp -File -Recurse | Move-Item -Destination .
+        # Verify digital signature against Google's public key
+        openssl dgst -sha384 -verify "${{ github.workspace }}/repo/build-aux/bouf/cng-release-signing-key.pem" -signature kmscng.msi.sig kmscng.msi
+        # Finally, install the CNG provider
+        msiexec /i kmscng.msi /qn /norestart
+
+    - name: Install rclone and pandoc
+      shell: pwsh
+      run: choco install pandoc rclone --no-progress -y
+
+    - name: Prepare Release Notes
+      shell: pwsh
+      run: |
+        # Release notes are just the tag body on Windows
+        cd repo
+        git tag -l --format='%(contents:body)' ${{ inputs.version }} > "${{ github.workspace }}/notes.rst"
+    
+    - name: 'Authenticate to Google Cloud'
+      uses: 'google-github-actions/auth@v1'
+      with:
+        workload_identity_provider: ${{ inputs.gcpWorkloadIdentityProvider }}
+        service_account: ${{ inputs.serviceAccountName }}
+
+    - name: Download old builds
+      shell: pwsh
+      env:
+        RCLONE_DRIVE_ENV_AUTH: 'true'
+        RCLONE_DRIVE_SHARED_WITH_ME: 'true'
+      run: |
+        rclone copy --transfers 40 :drive:old_builds "${{ github.workspace }}/old_builds"
```

Comment:
Shouldn't `robocopy` fulfil that need? Especially for local file operations, `rclone` as a third-party library seems unnecessary.

Reply:
rclone is used to download data from a remote storage system (google cloud in this case), robocopy is just for local copies.

Reply:
Ah my mistake, that's fine then.